### PR TITLE
Add support for favicon

### PR DESCRIPTION
**Pull request by isidentical**
Set favicon through `favicon` property as a file path relative to execution point. Send `favicon` on server ping action if defined as a base64 data.

### CODE REVIEW

**Comment by isidentical** [comment]
image

File: mcserver/objects/packet_handler.py [hunk 1]
```diff
@@ -118,10 +119,12 @@ def decode_plugin_message(cls, msg: ClientMessage):
                     msg.protocol_version,
                     "???"),
                 "protocol": msg.protocol_version
-            }
+            },
         }
-        # TODO: Server icon
+        if ServerCore.options["favicon"]:
```
Comment (Martmists-GH) [inline mcserver/objects/packet_handler.py]: This should use the `server.icon` file instead, similar to the official server implementation

File: mcserver/utils/misc.py [hunk 2]
```diff
@@ -44,7 +45,8 @@
     'prevent-proxy-connections': False,
     'use-native-transport': True,
     'enable-rcon': False,
-    'motd': 'A Minecraft Server'
+    'motd': 'A Minecraft Server',
+    'favicon': None
```
Comment (Martmists-GH) [inline mcserver/utils/misc.py]: This constant should reflect the default minecraft [server.properties](https://server.properties)

**Review by Martmists-GH — CHANGES_REQUESTED** [review]

File: mcserver/objects/packet_handler.py (hunk 1 above)
Comment (isidentical) [reply mcserver/objects/packet_handler.py]: got it.

File: mcserver/utils/misc.py [hunk 3]
```diff
@@ -65,7 +66,15 @@ def open_local(filename: str):
     dir_name = dirname(inspect.stack()[1].filename)
     return open(join(dir_name, filename))
 
-
+def read_favicon() -> str:
```
Comment (Martmists-GH) [inline mcserver/utils/misc.py]: This should be a Union[str, bool] now that we return False.

File: mcserver/utils/misc.py [hunk 4]
```diff
@@ -44,7 +45,7 @@
     'prevent-proxy-connections': False,
     'use-native-transport': True,
     'enable-rcon': False,
-    'motd': 'A Minecraft Server'
+    'motd': 'A Minecraft Server',
```
Comment (Martmists-GH) [inline mcserver/utils/misc.py]: Remove trailing comma

File: mcserver/utils/misc.py (hunk 3 above)
Comment (isidentical) [reply mcserver/utils/misc.py]: True

Comment (isidentical) [reply mcserver/utils/misc.py]: Better to return `None` and make it Optional[str]

**Pull request merged**